src/v/cluster/controller_probe.cc

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "cluster/controller_probe.h"
+
+#include "cluster/controller.h"
+#include "cluster/health_monitor_backend.h"
+#include "cluster/health_monitor_types.h"
+#include "cluster/members_table.h"
+#include "cluster/partition_leaders_table.h"
+#include "prometheus/prometheus_sanitize.h"
+#include "ssx/metrics.h"
+
+#include <seastar/core/metrics.hh>
+
+#include <absl/container/flat_hash_set.h>
+
+namespace cluster {
+
+controller_probe::controller_probe(controller& c) noexcept
+  : _controller(c) {
+    _controller._raft_manager.local().register_leadership_notification(
+      [this](
+        raft::group_id group,
+        model::term_id /*term*/,
+        std::optional<model::node_id> leader_id) {
+          // We are only interested in notifications regarding the controller
+          // group.
+          if (_controller._raft0->group() != group) {
+              return;
+          }
+
+          if (leader_id != _controller.self()) {
+              _public_metrics.reset();
+          } else {
+              setup_metrics();
+          }
+      });
+}
+
+void controller_probe::setup_metrics() {
+    namespace sm = ss::metrics;
+
+    if (config::shard_local_cfg().disable_metrics()) {
+        return;
+    }
+
+    _public_metrics = std::make_unique<ss::metrics::metric_groups>(
+      ssx::public_metrics_handle);
+    _public_metrics->add_group(
+      prometheus_sanitize::metrics_name("cluster"),
+      {
+        sm::make_gauge(
+          "brokers",
+          [this] {
+              const auto& members_table
+                = _controller.get_members_table().local();
+              return members_table.all_broker_ids().size();
```

Comment:
Is it worth adding a `members_table::all_brokers_count()`?

Reply:
Sure. Added it.